### PR TITLE
add tests and checksum validation for prebuilt native addons

### PR DESCRIPTION
**Pull request by rochdev**
### What does this PR do?

Add tests and checksum validation for prebuilt native addons.

### Motivation

A segmentation fault recently occurred in a beta version. After investigation, it was found that the cause was a corrupted binary. By adding tests that are using the prebuilt binaries and checksum validation, we ensure that we are not shipping corrupted binaries.

While adding the additional checks, it was also discovered that support for 32-bit operation systems was broken for Linux and Mac and this is fixed as well in this PR.

### Additional Notes

Sample run: https://circleci.com/workflow-run/727c3972-577c-43f3-8e9d-bb74239c658b

### CODE REVIEW

**Comment by brettlangdon** [comment]
We don't have tests for darwin, windows, or alpine?

**Comment by rochdev** [comment]
> We don't have tests for darwin, windows, or alpine?

Windows and Mac are currently only supported for development environments and not production.

Alpine is tested for ia32 since the only available ia32 Docker image for Node is Alpine-based, while x64 is tested on Debian. It's unlikely that just an architecture change would make a difference between glibc and musl, so this should be enough. In all cases, corruption would still be captured since the tests would not pass regardless of if on Debian or Alpine.

File: packages/dd-trace/src/platform/node/metrics.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ module.exports = function () {
         nativeMetrics = require('node-gyp-build')(path.join(__dirname, '..', '..', '..', '..', '..'))
         nativeMetrics.start()
       } catch (e) {
-        log.error('Unable to load native metrics module. Some metrics will not be available.')
+        log.error(e)
```
[the remark this answers is not in the document]
Comment (rochdev) [reply packages/dd-trace/src/platform/node/metrics.js]: The error messages for this are usually very descriptive. The main issue also is that I want to allow receiving the error object in the logger. Another option is to log twice.

File: scripts/prepublish.js [hunk 2]
```diff
@@ -185,6 +188,31 @@ function copyPrebuilds () {
...
+  })
+}
+
+function validatePrebuilds () {
+  platforms.forEach(platform => {
+    fs.readdirSync(path.join('prebuilds', platform))
+      .filter(file => /^node-\d+\.node$/.test(file))
+      .forEach(file => {
+        const content = fs.readFileSync(path.join('prebuilds', platform, file))
+        const sum = fs.readFileSync(path.join('prebuilds', platform, `${file}.sha1`), 'ascii')
```
[the remark this answers is not in the document]
Comment (rochdev) [reply scripts/prepublish.js]: Correct. If any file is missing it should throw.

**Pull request merged**